### PR TITLE
New align screen

### PR DESCRIPTION
**Pull request by germanferrero**
This PR introduces the v0.2.15 of the LimeApp, We have skipped v0.2.11 through v0.2.14 so all the new changes are:

firmware upgrade: add one click firmware upgrade
firstbootwizard: add do not ask again option to fbw banner
align: New align screen
remotesupport: Add support for handling and joining tmate sessions.

for more info checkout the [changelog](https://github.com/libremesh/lime-app/blob/develop/CHANGELOG.md)

The new align screen uses rpcd-mod-iwinfo ubus endpoints instead of lime-openairview.
All the functionality of lime-openairview except of spectral scan is removed. Which breaks compatibility with prior lime-app versions. That's why this PR includes changes + release bump altogether.

### CODE REVIEW

**Review by spiccinini — CHANGES_REQUESTED** [review]

**Comment by spiccinini** [comment]
@G10h4ck the bat-hosts format created by shared-state-publish_bat_hosts, doing `string.gsub(hostname.."_"..ifname, "%W", "_")` is hard deserialize back into `hostname` and `interface`. Are we using this format to be compatible somhow with the batman generation of the file? who was creating it before shared-state? alfred? 
What do you think about changing the format? This will be backward incompatible in a way that some nodes won't understand the shared-state data of the others, right?  What do you think about this?

**Comment by spiccinini** [comment]
I've pushed a commit with some but not all enhancements

File: packages/lime-system/files/usr/lib/lua/lime/utils.lua [hunk 1]
```diff
@@ -469,16 +485,20 @@ function utils.get_bathost(mac, outgoing_iface)
...
+				hostname, iface = utils.bathost_deserialize(line)
 			end
 			line = bathosts:read("*l")
 		end
 	end
-	if hostname == nil then
-		local ipv6ll = utils.mac2ipv6linklocal(mac) .. "%" .. outgoing_iface
-		hostname = utils.http_client_get("http://[" .. ipv6ll .. "]/cgi-bin/hostname", 10):gsub("\n", "")
+	if hostname == nil and outgoing_iface then
+		iface = outgoing_iface
```
Comment (germanferrero) [inline packages/lime-system/files/usr/lib/lua/lime/utils.lua:494]: We cannot say that the neighbor iface will be the same as our outgoing_iface
For instance in the case that wlan1_mesh of node A is associated with wlan2_mesh of node B.

Before, iface was not in the returned table. Do you have a better idea? :)

Comment (spiccinini) [reply packages/lime-system/files/usr/lib/lua/lime/utils.lua:494]: Yes indeed, I was confused on what was the data in the bat-hosts.

**Comment by G10h4ck** [comment]
> @G10h4ck the bat-hosts format created by shared-state-publish_bat_hosts, doing `string.gsub(hostname.."_"..ifname, "%W", "_")` is hard deserialize back into `hostname` and `interface`. Are we using this format to be compatible somhow with the batman generation of the file? who was creating it before shared-state? alfred?
> What do you think about changing the format? This will be backward incompatible in a way that some nodes won't understand the shared-state data of the others, right? What do you think about this?

AFAIR the format can be changed without breaking other nodes, the important thing is that we keep the information there so the names are unique. Also I remember batadv naming had some limitations on the characters that can be used, and at some point it did also some tricky like substityte `-` with `_` or the opposite or something similar.

File: packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils [hunk 2]
```diff
@@ -33,6 +34,21 @@ local function get_cloud_nodes(msg)
     return
 end
 
+local function get_bathost(msg)
+    mac = utils.shell_quote(msg.mac)
+    iface = utils.shell_quote(msg.outgoing_iface)
+    local bathost = utils.get_bathost(msg.mac, msg.outgoing_iface)
```
[the remark this answers is not in the document]
Comment (germanferrero) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils:42]: I've add is_valid_mac to lime.utils and used that

File: packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils [hunk 3]
```diff
@@ -33,6 +34,21 @@ local function get_cloud_nodes(msg)
     return
 end
 
+local function get_bathost(msg)
```
[the remark this answers is not in the document]
Comment (germanferrero) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils:37]: I think that we can generalize later on if we begin to query for hostnames that are not present in bat-host's shared state.
Also, if we decide to keep it tied to bathosts, we could move this to:
shared-state-bat_hosts/files/usr/libexec/rpcd/bat-hosts.lua
The code in lime.utils to:
shared-state-bat_hosts/files/usr/lib/lua/bat-hosts.lua
and the tests to
shared-state-bat_hosts/tests/test_bat-hosts.lua
 
 In that way we wouldn't be introducing a hidden dependency on shared-state-bat_hosts to lime utils

[the remark this answers is not in the document]
Comment (germanferrero) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils:37]: I've created https://github.com/libremesh/lime-packages/compare/master...germanferrero:new-align-screen-move-files to see how this refactoring would look like. I prefer it, to stop bloating lime-utils.

[the remark this answers is not in the document]
Comment (spiccinini) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils:37]: Great i also like it more!

[the remark this answers is not in the document]
Comment (spiccinini) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils:37]: There is an error in the line `     local bathost = utils.get_bathost(msg.mac, msg.outgoing_iface)` as it is still using utils.

[the remark this answers is not in the document]
Comment (germanferrero) [reply packages/ubus-lime-utils/files/usr/libexec/rpcd/lime-utils]: Solved and tested now

**Comment by germanferrero** [comment]
An upgrade to LimeApp v0.2.14 will be needed since get_bathost ubus endpoint was reallocated.
When the CI at the LimeApp repo ends up and https://github.com/libremesh/lime-packages/tree/lime-app-v0.2.14 gets available,
we will need to cherry-pick the single commit in that branch.

**Review by spiccinini — CHANGES_REQUESTED** [review]

**Comment by germanferrero** [comment]
@spiccinini I've updated this PR with LimeApp v0.2.15

**Review by spiccinini — APPROVED** [review]

**Review by nicopace — APPROVED** [review]

**Pull request merged**